tt-train/sources/ttml/modules/grouped_query_attention.hpp

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "autograd/module_base.hpp"
+#include "autograd/tensor.hpp"
+
+namespace ttml::modules {
+
+struct GQAConfig {
+    uint32_t embedding_dim{};
+    uint32_t num_heads{};
+    uint32_t num_groups{};
+    float dropout_prob{};
+    std::shared_ptr<autograd::ModuleBase> embedding{};
```

Comment:
I think embedding should not be a parto of config. I prefer when config could be easy serializable.